addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -114,52 +110,72 @@ private synchronized void login() {
 
-            InputStream response = connection.getInputStream();
-            String line = readResponse(response);
+            ContentResponse response = httpClient.newRequest(url)
+                    .method(HttpMethod.POST)
+                    .header(HttpHeader.ACCEPT_LANGUAGE, "en-US,en")
+                    .header(HttpHeader.ACCEPT_ENCODING, "gzip, deflate")
+                    .header("X-Requested-With", "XMLHttpRequest")
+                    .agent(TAHOMA_AGENT)
+                    .content(new StringContentProvider(urlParameters), "application/x-www-form-urlencoded; charset=UTF-8")
+                    .send();
 
-            SomfyTahomaLoginResponse data = gson.fromJson(line, SomfyTahomaLoginResponse.class);
+            SomfyTahomaLoginResponse data = gson.fromJson(response.getContentAsString(), SomfyTahomaLoginResponse.class);
 
             if (data.isSuccess()) {
-                String version = data.getVersion();
-                logger.debug("SomfyTahoma cookie: {}", cookie);
-                logger.debug("SomfyTahoma version: {}", version);
+                logger.debug("SomfyTahoma version: {}", data.getVersion());
                 updateStatus(ThingStatus.ONLINE);
             } else {
-                logger.debug("Login response: {}", line);
+                logger.debug("Login response: {}", response.getContentAsString());
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error logging in");
-                throw new SomfyTahomaException(line);
+                throw new SomfyTahomaException(response.getContentAsString());
             }
-        } catch (MalformedURLException e) {
-            logger.error("The URL '{}' is malformed!", url, e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    "The URL '" + url + "' is malformed: ");
-        } catch (Exception e) {
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
             logger.error("Cannot get login cookie!", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot get login cookie");
+        } catch (Exception e) {
+            logger.error("Cannot start http client", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot start http client");
         }
+
+    }
+
+    private ArrayList<SomfyTahomaEvent> getEvents() {
+        String url;
+        String line = "";
+
+        try {
+            url = TAHOMA_URL + "getEvents";
+
+            line = sendDataToTahomaWithCookie(url, "");
+            logger.debug("Events response: {}", line);
+            SomfyTahomaEvent[] array = gson.fromJson(line, SomfyTahomaEvent[].class);
+            return filterStateChangedEvents(array);
+        } catch (SomfyTahomaException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unauthorized");
```

Comment:
Unauthorized...

Maybe add that the user should check the credential on the thing.

Reply:
done. 
I've realized that Jetty http client throws an ExecutionException containing protocol violation message when login cookie to Somfy Tahoma cloud service expires (some http implementation incompatibility?). My former HttpsUrlConnection implementation returned 404 in this case, so I've had to add extra check for this behaviour where makes sense.